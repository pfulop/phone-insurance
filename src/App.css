.appHeader {
  text-align: center;
  background-color: #011c27;
  padding: 4px;
  color: white;
}

@media screen and (min-width: 480px) {
  .appHeader {
    margin-bottom: 16px;
    padding: 16px;
  }
}

.cardWrapper {
  display: flex;
  justify-content: center;
}
