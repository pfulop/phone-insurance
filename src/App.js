import React, { Component } from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter, Route } from 'react-router-dom';
import css from './App.css';
import createNewStore from './store';
import Home from './components/Home/Home';
import Form from './components/Claim/Form';
import Finish from './components/Finish/Finish';

const store = createNewStore();

class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <BrowserRouter>
          <div className={css.App}>
            <header className={css.appHeader}>
              <h1 className="App-title">Phone insurance claim</h1>
            </header>
            <div className={css.cardWrapper}>
              <Route path="/" exact component={Home} />
              <Route path="/start-claim" exact component={Form} />
              <Route path="/finish" exact component={Finish} />
            </div>
          </div>
        </BrowserRouter>
      </Provider>
    );
  }
}

export default App;
