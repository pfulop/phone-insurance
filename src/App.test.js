import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';
import { addAnswer, ADD_ANSWER } from './actions/answers';

describe('App', () => {
  it('should be defined', () => {
    expect(App).toBeDefined();
  });
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('should render correctly', () => {
    const tree = shallow(<App />);
    expect(tree).toMatchSnapshot();
  });

  it('should create action', () => {
    const expectedAction = {
      type: ADD_ANSWER,
      answer: { a: 'a' },
    };
    expect(addAnswer({ a: 'a' })).toEqual(expectedAction);
  });
});
