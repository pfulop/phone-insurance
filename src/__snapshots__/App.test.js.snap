// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <BrowserRouter>
        <div
                className={undefined}
        >
                <header
                        className={undefined}
                >
                        <h1
                                className="App-title"
                        >
                                Phone insurance claim
                        </h1>
                </header>
                <div
                        className={undefined}
                >
                        <Route
                                component={[Function]}
                                exact={true}
                                path="/"
                        />
                        <Route
                                component={[Function]}
                                exact={true}
                                path="/start-claim"
                        />
                        <Route
                                component={[Function]}
                                exact={true}
                                path="/finish"
                        />
                </div>
        </div>
</BrowserRouter>,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className={undefined}
>
          <header
                    className={undefined}
          >
                    <h1
                              className="App-title"
                    >
                              Phone insurance claim
                    </h1>
          </header>
          <div
                    className={undefined}
          >
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/"
                    />
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/start-claim"
                    />
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/finish"
                    />
          </div>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <header
              className={undefined}
>
              <h1
                            className="App-title"
              >
                            Phone insurance claim
              </h1>
</header>,
            <div
              className={undefined}
>
              <Route
                            component={[Function]}
                            exact={true}
                            path="/"
              />
              <Route
                            component={[Function]}
                            exact={true}
                            path="/start-claim"
              />
              <Route
                            component={[Function]}
                            exact={true}
                            path="/finish"
              />
</div>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h1
                className="App-title"
>
                Phone insurance claim
</h1>,
              "className": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Phone insurance claim",
                "className": "App-title",
              },
              "ref": null,
              "rendered": "Phone insurance claim",
              "type": "h1",
            },
            "type": "header",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
/>,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/start-claim"
/>,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/finish"
/>,
              ],
              "className": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/start-claim",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/finish",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <BrowserRouter>
          <div
                    className={undefined}
          >
                    <header
                              className={undefined}
                    >
                              <h1
                                        className="App-title"
                              >
                                        Phone insurance claim
                              </h1>
                    </header>
                    <div
                              className={undefined}
                    >
                              <Route
                                        component={[Function]}
                                        exact={true}
                                        path="/"
                              />
                              <Route
                                        component={[Function]}
                                        exact={true}
                                        path="/start-claim"
                              />
                              <Route
                                        component={[Function]}
                                        exact={true}
                                        path="/finish"
                              />
                    </div>
          </div>
</BrowserRouter>,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div
            className={undefined}
>
            <header
                        className={undefined}
            >
                        <h1
                                    className="App-title"
                        >
                                    Phone insurance claim
                        </h1>
            </header>
            <div
                        className={undefined}
            >
                        <Route
                                    component={[Function]}
                                    exact={true}
                                    path="/"
                        />
                        <Route
                                    component={[Function]}
                                    exact={true}
                                    path="/start-claim"
                        />
                        <Route
                                    component={[Function]}
                                    exact={true}
                                    path="/finish"
                        />
            </div>
</div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <header
                className={undefined}
>
                <h1
                                className="App-title"
                >
                                Phone insurance claim
                </h1>
</header>,
              <div
                className={undefined}
>
                <Route
                                component={[Function]}
                                exact={true}
                                path="/"
                />
                <Route
                                component={[Function]}
                                exact={true}
                                path="/start-claim"
                />
                <Route
                                component={[Function]}
                                exact={true}
                                path="/finish"
                />
</div>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h1
                  className="App-title"
>
                  Phone insurance claim
</h1>,
                "className": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Phone insurance claim",
                  "className": "App-title",
                },
                "ref": null,
                "rendered": "Phone insurance claim",
                "type": "h1",
              },
              "type": "header",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/"
/>,
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/start-claim"
/>,
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/finish"
/>,
                ],
                "className": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/start-claim",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/finish",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
